src/python/pants/backend/python/util_rules/local_dists.py

```diff
@@ -184,7 +184,10 @@ async def build_local_dists(
         Pex,
         PexRequest(
             output_filename="local_dists.pex",
-            requirements=PexRequirements(wheels),
+            # It is safe to not set the resolve_name because building this PEX will not make
+            # network calls, thanks to `--intransitive` and having the wheel sources present.
+            # TODO: is that actually true?
+            requirements=PexRequirements(wheels, resolve_name=None),
```

Comment:
@benjyw and I DMed a little about this, but neither of us have rigorously verified this claim. This TODO blocks the PR, and I'll look into it tomorrow

Reply:
Alright, I tested this out by adding `install_requires` to the `setup()` call in `local_dists_test.py`, then changing `pex.py` to only set `--no-pypi` for local_dists.pex. The test still succeeded, as @benjyw and I expected.

Reply:
Did you test for an install_requires that is sdist only, but uses a PEP-517 / PEP-518 build backend?

Reply:
I did not. Either way, because this PR does not impact `[python-repos]` at all, it is now irrelevant. I removed the misleading comment.

Reply:
Ok, great.

I looked more at that code and concur anyhow. Since the requirements are in fact all wheel file local paths, i.e.: locally built already, no network access should happen. It's only in the dists.py rules where that happens and indexes are potentially needed.